
--- a/Makefile.check	2012-09-24 17:51:17.911999997 +0400
+++ b/Makefile.check	2012-09-24 17:51:32.598000414 +0400
@@ -55,7 +55,7 @@
 	.objs/xmlconf_test testsuite/xml/xmlconf/xmlconf.xml --valid --not-wellformed
 	.objs/xmlcatconf_test testsuite/xml/xmlcatconf/xmlcatconf.xml
 
-league-valgrind:
+league-valgrind: league
 	$(VALGRIND) $(VALGRIND_FLAGS) .objs/library_level_test
 	$(VALGRIND) $(VALGRIND_FLAGS) --leak-check=full .objs/test_193
 
@@ -65,7 +65,7 @@
 	.objs/test_142
 	.objs/test_147
 
-sqlite3-valgrind:
+sqlite3-valgrind: sqlite3
 	$(VALGRIND) $(VALGRIND_FLAGS) .objs/test_147
 
 amf: amf-setup
