diff --git a/Makefile.build b/Makefile.build
index b40d399..6c230f5 100644
--- a/Makefile.build
+++ b/Makefile.build
@@ -128,6 +128,9 @@
 	$(GPRBUILD) $(GPRBUILD_FLAGS) -Pgnat/matreshka_amf_utp.gpr
 
 uaflex: league
+        # Disable debug information for the file due to gcc 7 bug
+	$(GPRBUILD) $(GPRBUILD_FLAGS) -f -c -g0 -Pgnat/matreshka_uaflex.gpr \
+          source/uaflex/matreshka-internals-finite_automatons.adb
 	$(GPRBUILD) $(GPRBUILD_FLAGS) -Pgnat/matreshka_uaflex.gpr
 
 # Disable parallel build, it is handled by gnatmake/gprbuild
