Index: cmof-internals-attribute_mappings.ads
===================================================================
--- cmof-internals-attribute_mappings.ads	(revision 1211)
+++ cmof-internals-attribute_mappings.ads	(working copy)
@@ -297,6 +297,7 @@
            Metamodel.MP_CMOF_Element_Owner                => 4,
            Metamodel.MP_CMOF_Named_Element_Qualified_Name => 5,
            Metamodel.MP_CMOF_Named_Element_Visibility     => 6,
+           Metamodel.MP_CMOF_Package_Uri                  => 7,
            others                                         => 0),
         Types.E_Package_Import =>
           (Metamodel.MP_CMOF_Package_Import_Imported_Package    => 1,
Index: cmof-internals-attributes.adb
===================================================================
--- cmof-internals-attributes.adb	(revision 1211)
+++ cmof-internals-attributes.adb	(working copy)
@@ -2040,4 +2040,44 @@
       end if;
    end Internal_Set_Visibility;
 
+   ----------------------
+   -- Internal_Get_Uri --
+   ----------------------
+
+   function Internal_Get_Uri
+     (Self : CMOF_Element) return AMF.AMF_String
+   is
+   begin
+      return
+        League.Strings.Internals.Create
+         (Elements.Table (Self).Member
+           (Member_Offset
+             (Elements.Table (Self).Kind,
+              MP_CMOF_Package_Uri)).String_Value);
+   end Internal_Get_Uri;
+
+   ----------------------
+   -- Internal_Set_Uri --
+   ----------------------
+
+   procedure Internal_Set_Uri
+     (Self : CMOF_Element;
+      To   : AMF.AMF_String) is
+   begin
+      Matreshka.Internals.Strings.Dereference
+       (Elements.Table (Self).Member
+         (Member_Offset
+           (Elements.Table (Self).Kind,
+            MP_CMOF_Package_Uri)).String_Value);
+      Elements.Table (Self).Member
+       (Member_Offset
+         (Elements.Table (Self).Kind,
+            MP_CMOF_Package_Uri)).String_Value := League.Strings.Internals.Get_Shared (To);
+      Matreshka.Internals.Strings.Reference
+       (Elements.Table (Self).Member
+         (Member_Offset
+           (Elements.Table (Self).Kind,
+            MP_CMOF_Package_Uri)).String_Value);
+   end Internal_Set_Uri;
+
 end Cmof.Internals.Attributes;
Index: cmof-internals-attributes.ads
===================================================================
--- cmof-internals-attributes.ads	(revision 1211)
+++ cmof-internals-attributes.ads	(working copy)
@@ -383,4 +383,11 @@
      (Self : CMOF_Element;
       To   : CMOF_Element);
 
+   function Internal_Get_Uri
+     (Self : CMOF_Element) return AMF.AMF_String;
+
+   procedure Internal_Set_Uri
+     (Self : CMOF_Element;
+      To   : AMF.AMF_String);
+
 end CMOF.Internals.Attributes;
Index: cmof-internals-reflection.adb
===================================================================
--- cmof-internals-reflection.adb	(revision 1211)
+++ cmof-internals-reflection.adb	(working copy)
@@ -1042,6 +1042,11 @@
          elsif Property = MP_CMOF_Named_Element_Visibility then
             return (Kind => AMF.Values.Value_None);
 
+         elsif Property = MP_CMOF_Package_Uri then
+            return
+             (AMF.Values.Value_String,
+              Internal_Get_Uri (Self));
+
          else
             raise Program_Error;
          end if;
@@ -1825,6 +1830,9 @@
          elsif Property = MP_CMOF_Named_Element_Visibility then
             null;
 
+         elsif Property = MP_CMOF_Package_Uri then
+            Internal_Set_Uri (Self, Value.String_Value);
+
          else
             raise Program_Error;
          end if;
Index: cmof-internals-constructors.adb
===================================================================
--- cmof-internals-constructors.adb	(revision 1224)
+++ cmof-internals-constructors.adb	(working copy)
@@ -754,6 +754,8 @@
                      4      => (M_Element, 0),             --  owner
                      5      => (M_String, Matreshka.Internals.Strings.Shared_Empty'Access),
                                                            --  qualifiedName
+                     7      => (M_String, Matreshka.Internals.Strings.Shared_Empty'Access),
+                                                           --  uri
                      others => (Kind => M_None)));
       Allocate_Collection_Of_Cmof_Element_Slots (Self, 12);
       Initialize_Set_Collection          --  elementImport
